### PR TITLE
add cross ref to htpdataset, fix cross ref and references on variant

### PR DESCRIPTION
**Pull request by sierra-moxon**
- adds various required attributes to variant objects
- adds cross references to HTP datasets.

### CODE REVIEW

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
- Added 3
           

Complexity increasing per file
==============================
- src/etl/variation_etl.py  17
         

Complexity decreasing per file
==============================
+ src/files/s3_file.py  -1
         

Clones added
============
- src/etl/htp_metadataset_etl.py  2
- src/etl/variation_etl.py  4
- src/etl/bgi_etl.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6170804&bid=20147973)

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
- Added 3
           

Complexity increasing per file
==============================
- src/etl/variation_etl.py  17
         

Complexity decreasing per file
==============================
+ src/files/s3_file.py  -1
         

Clones added
============
- src/etl/bgi_etl.py  1
- src/etl/variation_etl.py  4
- src/etl/htp_metadataset_etl.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6170804&bid=20147973)

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
- Added 3
           

Complexity increasing per file
==============================
- src/etl/variation_etl.py  17
         

Complexity decreasing per file
==============================
+ src/files/s3_file.py  -1
         

Clones added
============
- src/etl/bgi_etl.py  1
- src/etl/htp_metadataset_etl.py  2
- src/etl/variation_etl.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6170804&bid=20147973)

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
- Added 3
           

Complexity increasing per file
==============================
- src/etl/variation_etl.py  17
         

Clones added
============
- src/etl/bgi_etl.py  1
- src/etl/variation_etl.py  4
- src/etl/htp_metadataset_etl.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6170804&bid=20147973)

File: src/etl/variation_etl.py [hunk 1]
```diff
@@ -54,6 +54,19 @@ class VariationETL(ETL):
...
+        USING PERIODIC COMMIT %s
+        LOAD CSV WITH HEADERS FROM \'file:///%s\' AS row
+            MATCH (o:Variant {primaryKey:row.variantId})
+            MERGE (p:Publication {primaryKey:row.publicationId})
+              ON CREATE SET p.pubModId = row.pubModId,
+                 p.pubMedId = row.pubMedId,
+                 p.pubModUrl = row.pubModUrl,
+                 p.pubMedUrl = row.pubMedUrl
+
+           MERGE (o)-[pn:ASSOCIATION]-(p)
```
Comment (adamjohnwright) [inline src/etl/variation_etl.py:67]: This is just an idea, I feel like it should point towards (o) although I am not sure if it matters - just want to get it right before people start using it. This is the same as for other queries that you have made further down. Other than that looks great.

File: src/etl/variation_etl.py [hunk 2]
```diff
@@ -76,14 +89,52 @@ class VariationETL(ETL):
...
     """
+    notes_query_template = """
+           USING PERIODIC COMMIT %s
+           LOAD CSV WITH HEADERS FROM \'file:///%s\' AS row
+
+           MATCH (o:Variant {primaryKey:row.variantId})
+
+           CREATE (n:Note {primaryKey:row.noteId})
+                SET n.note = row.note          
+           MERGE (o)-[pn:ASSOCIATION]-(n)
```
Comment (adamjohnwright) [inline src/etl/variation_etl.py:100]: I am not sure but I think the pn variable is not necessary. Same thing with the next query

File: src/etl/variation_etl.py [hunk 3]
```diff
@@ -76,14 +89,52 @@ class VariationETL(ETL):
...
+                     id.localId = row.localId,
+                     id.crossRefCompleteUrl = row.crossRefCompleteUrl,
+                     id.prefix = row.prefix,
+                     id.crossRefType = row.crossRefType,
+                     id.uuid = row.uuid,
+                     id.page = row.page,
+                     id.primaryKey = row.primaryKey,
+                     id.displayName = row.displayName
+
+        MERGE (o)-[gcr:CROSS_REFERENCE]->(id)
```
Comment (adamjohnwright) [inline src/etl/variation_etl.py:135]: Variable "gcr" might not be necessary

**Review by adamjohnwright — APPROVED** [review]
I have left suggestions that I don't think are critical to pulling this in but could be considered. I am approving either way.

**Comment by sierra-moxon** [comment]
Thanks @adamjohnwright  - I can change all the xrefs in the datastore to the opposite direction if you want -- we have one method that sets (most) of the xref direction all the xrefs in the db (they point from dataId -> xrefId like these do).  At the moment, the downstream software does not use the direction of the relationship.  I'll put a case in to do that in another PR.

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
- Added 3
           

Complexity increasing per file
==============================
- src/etl/variation_etl.py  17
         

Clones added
============
- src/etl/htp_metadataset_etl.py  2
- src/etl/bgi_etl.py  1
- src/etl/variation_etl.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6170804&bid=20147973)

**Comment by adamjohnwright** [comment]
@sierra-moxon I have an intuition but I don't have a reason for the pointer to be in one direction or the other. Maybe we should discuss it more before you make a PR about it. Or we could discuss it in a PR. The directions of relationships seams like a small thing that we should probably revisit.

**Review by azurebrd — APPROVED** [review]

**Pull request merged**